src/Application/Messages/Commands/CreateMessageCommand/CreateMessageCommand.cs

```diff
@@ -0,0 +1,36 @@
+﻿using CoduTeam.Application.Common.Interfaces;
+using CoduTeam.Application.Messages.Commands.Common;
+using CoduTeam.Application.Messages.Mappers;
+using CoduTeam.Domain.Entities;
+
+namespace CoduTeam.Application.Messages.Commands.CreateMessageCommand;
+
+public record CreateMessageCommand(string Content) : BaseModifyMessageCommand(Content), IRequest
+{
+}
+public class CreateMessageComandHandler(IUser user, IApplicationDbContext dbContext) : IRequestHandler<CreateMessageCommand>
+{
+    public async Task Handle(CreateMessageCommand command, CancellationToken cancellationToken)
+    {
+        Guard.Against.Null(user.Id);
+
+        var message = command.ToMessage();
+
+        dbContext.Messages.Add(message);
+
+        var chat = await dbContext.Chats.FirstOrDefaultAsync(c => c.Id == message.ChatId, cancellationToken);
+
+        if (chat != null)
+        {
+            chat.Messages.Add(message);
+        }
+        else
+        {
+            var newChat = new Domain.Entities.Chat();
```

Comment:
Also incorrect, to create chat we should have at least 2 users , in this case it will be chat with yourself like 'Saved Messages'